lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+      setShowDropdown(false);
+    }
+  }
+
+  function handleFilterChange(value: string) {
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    onSearchTextChange(value);
+
+    if (value && value.trim() !== searchText.trim()) {
+      doSearch(value);
+      setPrevSearchText('');
+    }
+    else if (!value) {
+      setShowDropdown(false);
+    }
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+        if (elToFocus) {
+          setElToFocusId(elToFocus.id);
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          if (autoComplete && typeof elToFocus.textContent === 'string') {
+            onSearchTextChange(elToFocus.textContent);
+          }
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement;
+    switch (evt.key) {
+      case 'Enter':
+        if (typeof focusableBtnIndex === 'number' && elToFocusId) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        inputEle.setSelectionRange(searchText.length, searchText.length);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+        elToFocusId
+          ? (dropdownRef.current?.children[matches.length - 1].id === elToFocusId ? focusFirst() : focusNext())
+          : focusFirst();
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (!isEmpty) {
+          setShowDropdown(true);
+        }
+        elToFocusId ? (dropdownRef.current?.children[0].id === elToFocusId ? focusLast() : focusPrev()) :
+          focusLast();
+        evt.preventDefault();
+        break;
+      case 'Escape':
+        setShowDropdown(false);
+        setElToFocusId('');
+        setFocusableBtnIndex(null);
+        if (!showDropdown) {
+          handleFilterChange('');
+        }
+        break;
+    }
+  }
+
+  function doSearch(value: string) {
+    onSearch(value.trim());
+    setShowDropdown(true);
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  function setInlineOption() {
+    if (typeof matches[0].displayName === 'string' &&
+    matches[0].displayName.toLowerCase().indexOf(searchText.toLowerCase()) === 0) {
```

Comment:
Weird formatting here makes it hard to see that this is part of the `if` condition.  Maybe move the different parts of the condition out into local variables

Reply:
Updated here d99911ec4323048a2c2273180b1cbe2151ebf6da.